### PR TITLE
HAP-1395 - NuGet/DotNet is omitting the port in Artifactory URL

### PR DESCRIPTION
**Pull request by Or-Gabay**
- [x] All [tests](https://ci.appveyor.com/project/jfrog-ecosystem/build-info) passed. If this feature is not already covered by the tests, I added new tests.
-----

### CODE REVIEW

File: build-info-extractor-nuget/src/main/java/org/jfrog/build/extractor/nuget/drivers/ToolchainDriverBase.java [hunk 1]
```diff
@@ -68,6 +68,10 @@ protected String buildNugetSourceUrl(ArtifactoryBaseClient client, String repo)
                 .setScheme(rtUrl.getProtocol())
                 .setHost(rtUrl.getHost())
                 .setPath(rtUrl.getPath() + ARTIFACTORY_NUGET_API + repo);
+        int port = rtUrl.getPort();
+        if (port != -1){
```
Comment (eyalbe4) [inline build-info-extractor-nuget/src/main/java/org/jfrog/build/extractor/nuget/drivers/ToolchainDriverBase.java:72]: Missing white-space before {

**Review by eyalbe4 — CHANGES_REQUESTED** [review]

File: build-info-extractor-nuget/src/main/java/org/jfrog/build/extractor/nuget/drivers/ToolchainDriverBase.java [hunk 2]
```diff
@@ -68,6 +72,10 @@ protected String buildNugetSourceUrl(ArtifactoryBaseClient client, String repo)
                 .setScheme(rtUrl.getProtocol())
                 .setHost(rtUrl.getHost())
                 .setPath(rtUrl.getPath() + ARTIFACTORY_NUGET_API + repo);
+        int port = rtUrl.getPort();
+        if (port != -1) {
+            sourceUrlBuilder.setPort(port);
+        }
```
Comment (yahavi) [inline build-info-extractor-nuget/src/main/java/org/jfrog/build/extractor/nuget/drivers/ToolchainDriverBase.java:78]: The -1 check is redundant. Apache's URI builder already handles negative ports.
For our Apache HTTP client version see: [setPort](https://github.com/apache/httpcomponents-client/blob/rel/v4.5.12/httpclient/src/main/java/org/apache/http/client/utils/URIBuilder.java#L281) and [build](https://github.com/apache/httpcomponents-client/blob/rel/v4.5.12/httpclient/src/main/java/org/apache/http/client/utils/URIBuilder.java#L171).
For the latest Apache HTTP client code see [setPort](https://github.com/apache/httpcomponents-core/blob/master/httpcore5/src/main/java/org/apache/hc/core5/net/URIBuilder.java#L485) and [build](https://github.com/apache/httpcomponents-core/blob/master/httpcore5/src/main/java/org/apache/hc/core5/net/URIBuilder.java#L297).
```suggestion
                .setPath(rtUrl.getPath() + ARTIFACTORY_NUGET_API + repo)
                .setPort(rtUrl.getPort());

**Review by yahavi — CHANGES_REQUESTED** [review]

File: build-info-extractor-nuget/src/main/java/org/jfrog/build/extractor/nuget/drivers/ToolchainDriverBase.java (hunk 2 above)
Comment (Or-Gabay) [reply build-info-extractor-nuget/src/main/java/org/jfrog/build/extractor/nuget/drivers/ToolchainDriverBase.java:78]: Wow, thanks!.

**Comment by yahavi** [comment]
Resolves https://github.com/jfrog/jenkins-artifactory-plugin/issues/328

**Pull request merged**